### PR TITLE
[ticket/15859] Add core.viewforum_modify_topic_ordering

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-15859

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15859

### CODE REVIEW

**Comment by 3D-I** [comment]
!set Event

File: phpBB/viewforum.php [hunk 1]
```diff
@@ -285,6 +285,20 @@
 $sort_by_text = array('a' => $user->lang['AUTHOR'], 't' => $user->lang['POST_TIME'], 'r' => $user->lang['REPLIES'], 's' => $user->lang['SUBJECT'], 'v' => $user->lang['VIEWS']);
 $sort_by_sql = array('a' => 't.topic_first_poster_name', 't' => array('t.topic_last_post_time', 't.topic_last_post_id'), 'r' => (($auth->acl_get('m_approve', $forum_id)) ? 't.topic_posts_approved + t.topic_posts_unapproved + t.topic_posts_softdeleted' : 't.topic_posts_approved'), 's' => 'LOWER(t.topic_title)', 'v' => 't.topic_views');
 
+/**
+ * Modify the topic ordering if needed
+ *
+ * @event core.viewforum_modify_topic_ordering
+ * @var array	sort_by_text	Topic ordering options
+ * @var array	sort_by_sql		Topic ordering options
+ * @since 3.2.4
```
Comment (marc1706) [inline phpBB/viewforum.php]: As in the other PRs, please adjust this to `3.2.5-RC1`. Thanks.

File: phpBB/viewforum.php [hunk 2]
```diff
@@ -285,6 +285,20 @@
 $sort_by_text = array('a' => $user->lang['AUTHOR'], 't' => $user->lang['POST_TIME'], 'r' => $user->lang['REPLIES'], 's' => $user->lang['SUBJECT'], 'v' => $user->lang['VIEWS']);
 $sort_by_sql = array('a' => 't.topic_first_poster_name', 't' => array('t.topic_last_post_time', 't.topic_last_post_id'), 'r' => (($auth->acl_get('m_approve', $forum_id)) ? 't.topic_posts_approved + t.topic_posts_unapproved + t.topic_posts_softdeleted' : 't.topic_posts_approved'), 's' => 'LOWER(t.topic_title)', 'v' => 't.topic_views');
 
+/**
+ * Modify the topic ordering if needed
+ *
+ * @event core.viewforum_modify_topic_ordering
+ * @var array	sort_by_text	Topic ordering options
+ * @var array	sort_by_sql		Topic ordering options
```
Comment (marc1706) [inline phpBB/viewforum.php]: `Topic orderings options SQL equivalent`?

Comment (3D-I) [reply phpBB/viewforum.php]: `			$event['sort_by_sql'] = array(
				'a'	=> array('t.topic_priority', 't.topic_first_poster_name'),
				't'	=> array('t.topic_priority', 't.topic_last_post_time', 't.topic_last_post_id'),
				'r'	=> ( ($this->auth->acl_get('m_approve', $forum_id)) ? array('t.topic_priority', 't.topic_posts_approved + t.topic_posts_unapproved + t.topic_posts_softdeleted') : array('t.topic_priority', 't.topic_posts_approved') ),
				's'	=> array('t.topic_priority', 'LOWER(t.topic_title)'),
				'v'	=> array('t.topic_priority', 't.topic_views'),
			);`

File: phpBB/viewforum.php (hunk 1 above)
Comment (3D-I) [reply phpBB/viewforum.php]: Will do

File: phpBB/viewforum.php (hunk 2 above)
Comment (3D-I) [reply phpBB/viewforum.php]: @marc1706 I am overall prepending the newly created `topic_priority` to "cement" topics of any sort, due to an extension of mine.

Comment (marc1706) [reply phpBB/viewforum.php]: I understand that. My point was that it sounds to me like the description of sort_by_text and sort_by_sql should be different. While sort_by_text is a map between the options and their respective options, sort_by_sql is a map between the options and their SQL ordering equivalent. Hence why I suggested this change.

Comment (3D-I) [reply phpBB/viewforum.php]: Oh, I didn't got your point. I assumed you wanted to see the usage of it instead.

Pls, at this point feel free to amend the description if you so wish.

Comment (3D-I) [reply phpBB/viewforum.php]: I will do it, I see has been not yet rebased. ;))

**Comment by 3D-I** [comment]
Doc block amended as per request.

**Pull request merged**